### PR TITLE
[Dawn] Console Improvements

### PR DESCRIPTION
**Pull request by kevinmasd**
Addresses #295 

Modifies `Editor.js` and `JoyrideSteps.js`:
* Settings to allow users to change how large/small the console window is
* Button added to allow for mass copying (Untested in Windows)
* Joyride (Tour) updated to explain this

Modifies `webpack.config.babel.js`:
* Plugin addition conflicts with `package.json` information, preventing full debug capability.

### CODE REVIEW

File: dawn/renderer/components/Editor.js [hunk 1]
```diff
@@ -128,7 +131,7 @@ class Editor extends React.Component {
   }
 
   getEditorHeight(windowHeight) {
-    const windowNonEditorHeight = 231 + (this.props.showConsole * (this.consoleHeight + 40));
+    const windowNonEditorHeight = 235 + (this.props.showConsole * (this.state.consoleHeight + 40));
```
Comment (yizow) [inline dawn/renderer/components/Editor.js]: Is there a way to not have these hardcoded constants here? 
Some global variable? 

Not sure if this.state is similar.

File: dawn/renderer/components/Editor.js [hunk 2]
```diff
@@ -368,6 +371,22 @@ class Editor extends React.Component {
...
   }
 
+  raiseConsole() {
+    this.setState({ consoleHeight: this.state.consoleHeight + 10 }, () => {
+      this.CodeEditor.editor.resize();
+    });
+  }
+
+  lowerConsole() {
+    this.setState({ consoleHeight: this.state.consoleHeight - 10 }, () => {
```
Comment (yizow) [inline dawn/renderer/components/Editor.js]: Same here. define global constant "resize increment" or something

**Review by yizow — COMMENTED** [review]
Readability suggestions

File: dawn/renderer/components/JoyrideSteps.js [hunk 3]
```diff
@@ -29,13 +29,13 @@ export default [
     type: 'hover',
   }, {
     title: 'Code execution',
-    text: 'Use these buttons to run and stop your code, including a last-ditch Emergency-Stop.',
+    text: 'Use these buttons to run and stop your code, including a last-ditch Emergency-Stop. In addition to changing state,  you can simulate competition timings.',
```
Comment (kevinji) [inline dawn/renderer/components/JoyrideSteps.js]: Nit: space before "you". Also, what does "simulate competition timings" mean?

File: dawn/renderer/components/Editor.js [hunk 4]
```diff
@@ -482,6 +502,26 @@ class Editor extends React.Component {
               onClick={this.props.onClearConsole}
               glyph="remove"
             />
+            <TooltipButton
+              id="raise-console"
+              text="Raise Console"
+              onClick={this.raiseConsole}
+              glyph="arrow-up"
+              disabled={this.state.consoleHeight > 350}
```
Comment (kevinji) [inline dawn/renderer/components/Editor.js]: Can you give this constant a name? Looks like a magic number right now.

File: dawn/renderer/components/Editor.js [hunk 5]
```diff
@@ -482,6 +502,26 @@ class Editor extends React.Component {
...
+            />
+            <TooltipButton
+              id="lower-console"
+              text="Lower Console"
+              onClick={this.lowerConsole}
+              glyph="arrow-down"
+              disabled={this.state.consoleHeight < 100}
```
Comment (kevinji) [inline dawn/renderer/components/Editor.js]: Ditto.

File: dawn/renderer/components/Editor.js [hunk 6]
```diff
@@ -369,6 +373,22 @@ class Editor extends React.Component {
...
+  }
+
+  lowerConsole() {
+    this.setState({ consoleHeight: this.state.consoleHeight - windowInfo.UNIT }, () => {
+      this.CodeEditor.editor.resize();
+    });
+  }
+
+  copyConsole() {
+    clipboard.writeText(this.props.consoleData.join(''));
```
Comment (kevinji) [inline dawn/renderer/components/Editor.js:389]: No newline joins?

**Review by kevinji — APPROVED** [review]
Looks okay besides the minor concerns I mentioned.

File: dawn/renderer/components/Editor.js (hunk 6 above)
Comment (kevinmasd) [reply dawn/renderer/components/Editor.js:389]: Text already has \n sent from Runtime.

**Pull request merged**